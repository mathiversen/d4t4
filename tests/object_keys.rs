use data_parser::{parse, Result};
use indoc::indoc;
use insta::assert_json_snapshot;

#[test]
fn it_can_parse_different_keys() -> Result<()> {
    let markup = indoc!(
        r#"{
            _name: "Mr. Sunesson",
            $name: "Mr. Birgersson",
            0: "Mr. Karlsson",
            1-name: "Mr. Eriksson",
            2_name: "Mr. Fredriksson",
            name-1: "Mr. Andersson",
            name_2: "Mr. Johansson",
            "name-3": "Mr. Davidsson",
        }"#
    );
    let x = parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}

#[test]
fn it_errors_on_same_key() {
    let markup = indoc!(
        r#"{
            name: "Mr. Karlsson",
            name: "Mr. Eriksson",
        }"#
    );
    let x = parse(markup).err().unwrap();
    assert_eq!(format!("{}", x), "Object already contains key: name");
}
