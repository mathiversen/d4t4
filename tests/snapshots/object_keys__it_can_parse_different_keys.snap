---
source: tests/object_keys.rs
expression: x
---
{
  "$name": "Mr. Birgersson",
  "0": "Mr. Karlsson",
  "1-name": "Mr. Eriksson",
  "2_name": "Mr. Fredriksson",
  "_name": "Mr. Sunesson",
  "name-1": "Mr. Andersson",
  "name-3": "Mr. Davidsson",
  "name_2": "Mr. Johansson"
}
